lib/tasks/instrumentation_generator/templates/dependency_detection.tt

```diff
@@ -0,0 +1,29 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+
+require_relative '<%= @name.downcase %>/instrumentation'
+require_relative '<%= @name.downcase %>/chain'
+require_relative '<%= @name.downcase %>/prepend'
+
+DependencyDetection.defer do
+  named :<%= @name.downcase %>
+
+  depends_on do
+    # The class that needs to be defined to prepend/chain onto. This can be used
+    # to determine whether the library is installed.
+    defined?(::<%= @name.capitalize %>)
+    # Add any additional requirements to verify whether this instrumentation
+    # should be installed
+  end
+
+  executes do
+    ::NewRelic::Agent.logger.info "Installing <%= @name.capitalize %> instrumentation"
+
+    if use_prepend?
+      prepend_instrument ::<%= @name.capitalize %>, NewRelic::Agent::Instrumentation::<%= @name.capitalize %>::Prepend
+    else
+      chain_instrument NewRelic::Agent::Instrumentation::<%= @name.capitalize %>
```

Comment:
Hmm... thinking about my preference for something like `#classify` over `#capitalize`, perhaps we could just perform the operation once and stick it in an instance var like `@class_name`

Reply:
Ooh, that's a nice idea!